hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/common/AbstractStreamWriteFunction.java

```diff
@@ -194,8 +194,11 @@ public void setOperatorEventGateway(OperatorEventGateway operatorEventGateway) {
 
   private void restoreWriteMetadata() throws Exception {
     boolean eventSent = false;
+    HoodieTimeline pendingTimeline = this.metaClient.getActiveTimeline().filterPendingExcludingCompaction();
     for (WriteMetadataEvent event : this.writeMetadataState.get()) {
-      if (Objects.equals(this.currentInstant, event.getInstantTime())) {
+      // Must filter out the completed instants in case it is a partial failover,
```

Comment:
If I understand correctly, if the instant to recommit is rolled back by another process (for example, because its heartbeat expired), it won't be recommitted and the data within that instant will be lost, is that right?